File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int> mp;
        for(int i=0;i<nums.size();i++){
            int fi=target-nums[i];
            if(mp.find(fi)!=mp.end()){
                return {i,mp[fi]};
            }
            mp[nums[i]]=i;
        }
        return {-1,-1};
    }
};

File: 0001-two-sum/0001-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    let n=nums.length;
    const freq=new Map();
    for(let i=0;i<n;i++){
        if(freq.has(target-nums[i])){
            return [freq.get(target-nums[i]),i];
        }
        freq.set(nums[i],i);
    }
    return [-1,-1];
};